pkg/apis/streaming/v1alpha1/processor_defaults.go

```diff
@@ -44,4 +47,14 @@ func (s *ProcessorSpec) Default() {
 			s.Outputs[i].Alias = s.Outputs[i].Stream
 		}
 	}
+
+	if s.Template == nil {
+		s.Template = &corev1.PodSpec{}
+	}
+	if len(s.Template.Containers) == 0 {
+		s.Template.Containers = append(s.Template.Containers, corev1.Container{})
+	}
+	if s.Template.Containers[0].Name == "" {
```

Comment:
Should we always overwrite?

Reply:
we could, but why?

Reply:
it would be better as a validation constraint in that case

Reply:
Agreed. I think the container names are _implicitly_ part of the contract, so should be enforced.